docs/http_api.md

```diff
@@ -0,0 +1,114 @@
+# SQLD HTTP API
+
+This is the documentation for the sqld HTTP API.
+
+## Usage
+
+### The `Value` type
+
+The `Value` type represents an SQLite value. It has 4 variants:
+
+- Text: a UTF-8 encoded string
+- Integer: a 64-bit signed integer
+- Real: a 64-bits floating number
+- Blob: some binary data, encoded in base64
+- Null: the null value.
+
+All these types map to JSON straightforwardly, with two exceptions:
+- Blob: they are represented as an object with { "type": "blob", "base64": /* base64 encoded blob */}
+- Integers: they are represented as string since some languages are not able to represent them in full fidelity.
+
+### Routes
+
+#### Queries
+
+```
+POST /query
+```
+
+This endpoint supports sending batches of queries to the database. All of the statements in the batch are executed as part of a transaction. If any statement in the batch fails, an error is returned and the transaction is aborted, resulting in no change to the database.
+
+The HTTP API is stateless, which means that interactive transactions are not possible. Since all batches are executed as part of transactions, any transaction statements (e.g `BEGIN`, `END`, `ROLLBACK`...) are forbidden and will yield an error.
+
+##### Body
+
+The body for the query request has the following format:
+
+```
+type QueryBody = {
+    statements: Array<Query>
+}
+
+type Query = string | ParamQuery;
+type ParamQuery = { q: string, params: null | Record<string, Value> | Array<Value> }
```

Comment:
What does null mean here?  Or can we just make the params optional, as they are now in the client TS API?

Reply:
In the sense that they are optional, yes

Reply:
OK, works for me.

Reply:
What would you call it when a field is not defined?